File: 프로그래머스/0/120824. 짝수 홀수 개수/짝수 홀수 개수.py
def solution(num_list):
    answer = [0, 0]  # 짝수 개수와 홀수 개수를 저장할 리스트

    for n in num_list:  # 리스트의 각 원소를 순회
        answer[n % 2] += 1  # 짝수면 answer[0] 증가, 홀수면 answer[1] 증가

    return answer  # [짝수 개수, 홀수 개수] 반환

# 테스트 예제 실행
print(solution([1, 2, 3, 4, 5]))    # [2, 3]
print(solution([1, 3, 5, 7]))       # [0, 4]

File: 프로그래머스/0/120824. 짝수 홀수 개수/짝수 홀수 개수.java
public class Solution {

    public static int[] solution(int[] numList) {
        int[] answer = new int[2];  // answer[0]: 짝수 개수, answer[1]: 홀수 개수

        for (int i = 0; i < numList.length; i++) {
            int n = numList[i];
            answer[n % 2]++;  // n % 2가 0이면 짝수 → answer[0], 1이면 홀수 → answer[1]
        }

        return answer;
    }

    public static void main(String[] args) {
        int[] result1 = solution(new int[]{1, 2, 3, 4, 5});
        System.out.println("[" + result1[0] + ", " + result1[1] + "]");  // [2, 3]

        int[] result2 = solution(new int[]{1, 3, 5, 7});
        System.out.println("[" + result2[0] + ", " + result2[1] + "]");  // [0, 4]
    }
}
